lib/Documentable/Search.pm6

```diff
@@ -45,10 +56,35 @@ class Documentable::Search {
         @entries
     }
 
+    method calculate-category(@docs, $kind) {
+        if @docs.elems == 1 {
+            return @docs[0].categories[0];
+        } else {
+            return 'Operators' if @docs.map(*.categories).map(* ~~ /'operators'/).any;
+        }
+
+        given $kind {
+            when 'routine' { 'Routines' }
+            when 'syntax'  { 'Syntax'  }
+            default { die $kind }
+        }
+    }
+
+    method consume-entries-by-kind(@entries, $registry, $kind, $category) {
+        @entries.append: $registry.lookup($kind.Str, :by<kind>).map(-> $doc {
+            self.search-entry(:$category, value => $doc.name, url => $doc.url)
+        }).Slip;
+    }
+
     method search-entry(Str :$category, Str :$value, Str :$url is copy) {
+        if %!seen{"$category - $value"}:exists {
```

Comment:
OK, this looks for duplicate categories. That's a good thing; however, it's caused tests to fail (maybe), so it's probably something that should be done in a separate PR. I've created #161 but also Raku/doc#1912 is there from some time (you might remember it) so, even if this only creates a warning, I have a hunch it's related to the fails in CircleCI. And at any rate, since it involves two repositories and it's out of scope in this PR, I would really appreciate if you created a different PR *with this only* so that we can solve it properly before proceeding with search category streamlinization later.

Reply:
If there is already a test in the doc repo, that's great! I am not sure if the script actually got in the repo though, cannot find it at a glance. But yes, it makes sense to move it there.